src/ray/raylet/node_manager.cc

```diff
@@ -1584,48 +1584,53 @@ void NodeManager::HandleRequestWorkerLease(const rpc::RequestWorkerLeaseRequest
     worker_pool_.PrestartWorkers(task_spec, request.backlog_size(), available_cpus);
   }
 
-  if (!(RayConfig::instance().gcs_actor_scheduling_enabled() &&
-        task.GetTaskSpecification().IsActorCreationTask())) {
-    cluster_task_manager_->QueueAndScheduleTask(task, reply, send_reply_callback);
+  if (!request.grant_or_reject()) {
+    cluster_task_manager_->QueueAndScheduleTask(task, false, reply, send_reply_callback);
     return;
   }
 
-  auto send_reply_callback_wrapper = [this, actor_id, reply, send_reply_callback](
+  auto send_reply_callback_wrapper = [this, is_actor_creation_task, actor_id, reply,
+                                      send_reply_callback](
                                          Status status, std::function<void()> success,
                                          std::function<void()> failure) {
     if (!reply->rejected()) {
       send_reply_callback(status, success, failure);
       return;
     }
 
-    // If the reqiured resource and normal task resource exceed available resource,
-    // reject it.
-    ResourceSet normal_task_resources = cluster_task_manager_->CalcNormalTaskResources();
-    RAY_LOG(DEBUG) << "Reject leasing as the raylet has no enough resources."
-                   << " actor_id = " << actor_id
-                   << ", normal_task_resources = " << normal_task_resources.ToString()
-                   << ", local_resoruce_view = "
-                   << cluster_resource_scheduler_->GetLocalResourceViewString();
-    auto resources_data = reply->mutable_resources_data();
-    resources_data->set_node_id(self_node_id_.Binary());
-    resources_data->set_resources_normal_task_changed(true);
-    auto &normal_task_map = *(resources_data->mutable_resources_normal_task());
-    normal_task_map = {normal_task_resources.GetResourceMap().begin(),
-                       normal_task_resources.GetResourceMap().end()};
-    resources_data->set_resources_normal_task_timestamp(absl::GetCurrentTimeNanos());
+    if (is_actor_creation_task) {
```

Comment:
Why don't we follow this for normal tasks as well? Seems that would simplify the logic

Reply:
I don’t follow this to make it explicit that for normal task, we don’t use that normal task resources feedback loop. That only used by gcs actor scheduling.

Reply:
Hmm I agree with @ericl here. Would it work if we changed this to be agnostic to actor creation tasks vs normal tasks? I think we just need two flags from the client here:

- grant_or_reject (always true for GCS)
- include_current_resources (always true for GCS to update resources view)

My higher level question here is if there's a known issue if we queue actor creation tasks instead of rejecting immediately?

Reply:
Going to always queue actor creation tasks as well to be consistent with normal tasks. We can still reject once they are queued.